pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -796,7 +828,9 @@ func (r *ReconcileBareMetalHost) actionManageReady(prov provisioner.Provisioner,
 		info.host.ClearError()
 		return actionContinue{provResult.RequeueAfter}
 	}
-
+	if hasInspectAnnotation(info.host) {
+		return r.actionInspecting(prov, info)
+	}
```

Comment:
No, we need to switch the state not just start executing another state. We should do that in the state machine's `handleReady()` method.

Reply:
I'm not 100% sure if that's OK to create a new type (`actionInspection`), or if the name of the type is not confusing/overlapping with `actionInspecting`.

Reply:
fixed it